src/Microsoft.ML.TimeSeries/SequentialAnomalyDetectionTransformBase.cs

```diff
@@ -557,98 +538,93 @@ protected override sealed void InitializeStateCore()
         }
 
-        /// <summary>
-        /// Schema implementation to add slot name metadata to the produced output column.
-        /// </summary>
-        private sealed class SchemaImpl : ISchema
-        {
-            private readonly ISchema _parent;
-            private readonly int _col;
-            private readonly ColumnType _type;
-            private readonly string[] _names;
-            private readonly int _namesLength;
-            private readonly MetadataUtils.MetadataGetter<VBuffer<ReadOnlyMemory<char>>> _getter;
+        protected override IRowMapper MakeRowMapper(ISchema schema) => new Mapper(Host, this, schema);
 
-            public int ColumnCount { get { return _parent.ColumnCount; } }
+        public sealed class Mapper : IRowMapper
+        {
+            private readonly IHost _host;
+            private readonly SequentialAnomalyDetectionTransformBase<TInput, TState> _parent;
+            private readonly ISchema _parentSchema;
+            private readonly string[] _slotNames;
 
-            /// <summary>
-            /// Constructs the schema.
-            /// </summary>
-            /// <param name="schema">The schema we will wrap.
-            /// Aside from presenting that additional piece of metadata, the constructed schema
-            /// will appear identical to this input schema.</param>
-            /// <param name="col">The column in <paramref name="schema"/> that has the metadata.</param>
-            /// <param name="names"></param>
-            /// <param name="length"></param>
-            public SchemaImpl(ISchema schema, int col, string[] names, int length)
+            public Mapper(IHostEnvironment env, SequentialAnomalyDetectionTransformBase<TInput, TState> parent, ISchema inputSchema)
             {
-                Contracts.Assert(length > 0);
-                Contracts.Assert(Utils.Size(names) >= length);
-                Contracts.AssertValue(schema);
-                Contracts.Assert(0 <= col && col < schema.ColumnCount);
-                _parent = schema;
-                _col = col;
-
-                _names = names;
-                _namesLength = length;
-
-                _type = new VectorType(TextType.Instance, _namesLength);
-                Contracts.AssertValue(_type);
-                _getter = GetSlotNames;
+                Contracts.CheckValue(env, nameof(env));
+                _host = env.Register(nameof(Mapper));
+                _host.CheckValue(inputSchema, nameof(inputSchema));
+                _host.CheckValue(parent, nameof(parent));
+
+                _parent = parent;
+                _parentSchema = inputSchema;
+                _slotNames = new string[4];
+                _slotNames[0] = "Alert";
+                _slotNames[1] = "Raw Score";
+                _slotNames[2] = "P-Value Score";
+                _slotNames[3] = "Martingale Score";
```

Comment:
I would simplify and moved to _slotName definition.

Reply:
tried but getting build error: "... a field initialiser outside the constructor"

---
In reply to: [225357735](https://github.com/dotnet/machinelearning/pull/1254#discussion_r225357735) [](ancestors = 225357735)

Reply:
yep, tried this and got build error. 

---
In reply to: [226785830](https://github.com/dotnet/machinelearning/pull/1254#discussion_r226785830) [](ancestors = 226785830,226489440,225357735)

Reply:
oh u mean inside the constructor. ok

---
In reply to: [226806798](https://github.com/dotnet/machinelearning/pull/1254#discussion_r226806798) [](ancestors = 226806798,226785830,226489440,225357735)